gatorgrade/input/set_up_shell.py

```diff
@@ -0,0 +1,28 @@
+"""Set-up the shell commands."""
+import os, sys
+
+
+def run_setup(front_matter):
+    """Run the shell set up commands and exit the program if a command fails.
+
+    Args:
+        front_matter: A dictionary whose 'setup' key contains the set up commands
+        as a multi-line string.
+
+    """
+    print("Running set up commands...")
+    for line in front_matter["setup"].splitlines():
+        # Trims the white space
+        command = line.strip()
+        exit_status = os.system(command)
+        # Extracts the exit code number from the exit status.
+        exit_code = os.WEXITSTATUS(exit_status)
+        if exit_code != 0:
+            print(
+                f'The set up command "{command}" failed.\
+            Exiting GatorGrader.',
+                file=sys.stderr,
+            )
+            # If a set up command failed, exit the execution
+            # because environment was not set up correctly.
+            sys.exit()
```

Comment:
I think you may want to exit with an error code here, something like `sys.exit(1)`. That way a GitHub Actions run would fail, and other tools that expect well-defined exit code behavior will work correctly!

Reply:
Oh, I did not know that. Thanks for the advice